Spigot-Server-Patches/0230-LivingEntity-setKiller.patch

```diff
@@ -0,0 +1,30 @@
+From f6ced6606afb1efd02705d46f52280d4f4732c07 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Mon, 31 Jul 2017 01:49:48 -0500
+Subject: [PATCH] LivingEntity#setKiller
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+index 3c3e8b87..640ade4f 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+@@ -293,6 +293,16 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
+         return getHandle().killer == null ? null : (Player) getHandle().killer.getBukkitEntity();
+     }
+ 
++    // Paper start
++    @Override
++    public void setKiller(Player killer) {
++        EntityPlayer entityPlayer = killer == null ? null : ((CraftPlayer) killer).getHandle();
++        getHandle().killer = entityPlayer;
++        getHandle().lastDamager = entityPlayer;
++        getHandle().lastDamageByPlayerTime = entityPlayer == null ? 0 : 100;
```

Comment:
Ok, the 100 value is from net.minecraft.server.EntityLiving#damageEntity()
My copy has it at line 909, but i'm a bit outdated
https://gyazo.com/9a11f5178a3d312f53a14153eb0f0bdc

Its also used at line 314 in the same file for the getExpReward() method
https://gyazo.com/10cb084b2fe2243b9a8df4104a5be551
so, the PR has 100 set if a plugin uses the API to set a killer, and then to 0 if they are unsetting (setting null) a killer

Also same file in the die() method it checks this value to determine dropped items
https://gyazo.com/fc4ef3bd28ef80dd6a6fa3e9e1b692dc
https://gyazo.com/f3afa5af15239f1848bf6dbf97e5333f

And here's the obfuscated method Y() that ticks this value down to 0 (if its above 0) and sets the killer as null when the value reaches 0
https://gyazo.com/de2254e06040b20e308ee2e20ba0e37d